idea/src/org/jetbrains/kotlin/idea/inspections/RedundantProgressionStepInspection.kt

```diff
@@ -0,0 +1,89 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jetbrains.kotlin.idea.inspections
+
+import com.intellij.codeInsight.FileModificationService
+import com.intellij.codeInspection.*
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiElementVisitor
+import com.intellij.psi.impl.source.tree.LeafPsiElement
+import org.jetbrains.kotlin.idea.caches.resolve.analyze
+import org.jetbrains.kotlin.idea.core.replaced
+import org.jetbrains.kotlin.idea.imports.importableFqName
+import org.jetbrains.kotlin.lexer.KtTokens
+import org.jetbrains.kotlin.psi.KtBinaryExpression
+import org.jetbrains.kotlin.psi.KtConstantExpression
+import org.jetbrains.kotlin.psi.KtVisitorVoid
+import org.jetbrains.kotlin.psi.psiUtil.endOffset
+import org.jetbrains.kotlin.psi.psiUtil.startOffset
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.resolve.calls.callUtil.getResolvedCall
+
+class RedundantProgressionStepInspection : AbstractKotlinInspection() {
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean, session: LocalInspectionToolSession): PsiElementVisitor {
+        return object : KtVisitorVoid() {
+            override fun visitBinaryExpression(expression: KtBinaryExpression) {
+                if (isStepOneCall(expression)) {
+                    holder.registerProblem(expression,
+                                           "Iteration step is redundant. Recommended to remove it.",
+                                           ProblemHighlightType.WEAK_WARNING,
+                                           TextRange(expression.operationReference.startOffsetInParent, expression.endOffset - expression.startOffset),
+                                           EliminateStepOneFix())
+                }
+            }
+        }
+    }
+
+
+    class EliminateStepOneFix : LocalQuickFix {
+        override fun getName() = "Eliminate redundant iteration step size"
+
+        override fun getFamilyName() = name
+
+        override fun applyFix(project: Project, descriptor: ProblemDescriptor) {
+            if (!FileModificationService.getInstance().preparePsiElementForWrite(descriptor.psiElement)) return
+            val expression = descriptor.psiElement as? KtBinaryExpression ?: return
+            if (!isStepOneCall(expression)) return
+            expression.replaced(expression.left!!)
+        }
+    }
+
+    companion object {
+        private val STEP_FUNCTION_FQ_NAME = "kotlin.ranges.step"
+
+        fun isStepOneCall(expression: KtBinaryExpression): Boolean {
+            val left = expression.left as? KtBinaryExpression ?: return false
```

Comment:
You should use KtPsiUtil.deparenthesize() to handle the case when the range expression is in parentheses.